src/indicators/flooding/glofas/utils.py

```diff
@@ -180,10 +201,25 @@ def get_crps(
     return df_crps
 
 
-def get_groups_above_threshold(observations, threshold, min_duration=1):
-    groups = np.where(np.diff(observations > threshold, prepend=False, append=False))[
-        0
-    ].reshape(-1, 2)
+def get_groups_above_threshold(
+    observations: ArrayLike,
+    threshold: float,
+    min_duration: int = 1,
+    additional_condition: ArrayLike = None,
+) -> List:
+    """
+    Get indices where consecutive values are equal to or above a threshold
+    :param observations: The array of values to search for groups (length N)
+    :param threshold: The threshold above which the values must be
+    :param min_duration: The minimum group size (default 1)
+    :param additional_condition: (optional) Any additional condition the values must satisfy
```

Comment:
Nice!

Reply:
Thanks! We should really do this for every method, oh boy...